sympy/integrals/meijerint.py

```diff
@@ -1681,7 +1681,7 @@ def tr(p):
         # The antiderivative is most often expected to be defined
         # in the neighborhood of  x = 0.
         place = 0
-        if b < 0 or f.subs(x, 0).has(nan, zoo):
+        if not b.is_extended_nonnegative or f.subs(x, 0).has(nan, zoo):
```

Comment:
What is the intention here for non-real values of b or for the case where `b.is_extended_nonnegative` returns None (e.g. if b is `Symbol('b')`)?

Reply:
I **think** it is better to have `place = None` instead of `place = 0` for that case. Just by intuition.

Reply:
I don't know myself what this code should do. I just ask because it's not clear if you intend for the new code to be equivalent (given discussions elsewhere) or not. If it should be equivalent for real b then it should be:
```
if b.is_extended_negative or f.subs...
```
The previous version would raise if b is non-real or whenever the condition doesn't evaluate. The new version you add here allows both cases through (since `not None -> True`).

Reply:
Correct. I assumed that better to go with `place=None` than `place=0` if we cannot tell. 

Not sure what is required to get a complex value in there, but in general I would say that it is better to not raise errors in that case, #17045 is a motivating example.

Reply:
`b` is the exponent in a `a*x**b` expression. So the question is if one should do `hyperexpand` at 0 or not, given the information on `b`. Seems to me that the safest is to assume not do it at 0 if we do not know that the exponent is nonnegative.

Reply:
Okay then I think what you have here is fine. However I would personally prefer to spell it like:
```
    if b.is_extended_nonnegative is not True or ...
```
The reason I prefer this is that it suggests (to me at least) that you have considered the three cases in the ternary logic. Whenever I see `not x.is_y` I find myself unsure if the author really considered all three cases (often it seems they haven't).

Then again the docstring for BooleanTrue suggests against `is True` (and therefore disagrees with me).

In any case probably the clearest way is to turn the conditions around and avoid the ternary negation altogether:
```
if b.is_extended_nonnegative and not f.subs(...).has(zoo, oo):
    place = 0  # Assume we can expand at zero
else:
    place = None
```

Reply:
And, yes, this doesn't throw any error doing `integrate(cos(x**(2+I))/x**(2+I), x, meijerg=True)` as opposed to current master.

But complex constants are a problem (current master):

```
integrate(1/x**(2*I+1), x)
Out[15]: 
   -2⋅ⅈ
ⅈ⋅x    
───────
   2   

integrate(1/x**(2*I+1), x, meijerg=True)
Out[14]: 
⎧                          -2⋅ⅈ                                               
⎪                         x    ⋅Γ(-2⋅ⅈ)                                       
⎪                         ─────────────                           for │x│ > 1 ∨ │x│ < 1
⎪                           Γ(1 - 2⋅ⅈ)                                        
⎨                                                                             
⎪╭─╮1, 1 ⎛ 1    1 - 2⋅ⅈ │  ⎞   ╭─╮0, 2 ⎛1 - 2⋅ⅈ, 1          │  ⎞              
⎪│╶┐     ⎜              │ x⎟ + │╶┐     ⎜                    │ x⎟        otherwise
⎪╰─╯2, 2 ⎝-2⋅ⅈ     0    │  ⎠   ╰─╯2, 2 ⎝            -2⋅ⅈ, 0 │  ⎠              
⎩                                                                             
```